base.js

```diff
@@ -6,6 +6,28 @@ const _ = require('lodash')
 const async = require('async')
 const EventEmitter = require('events')
 
+const extractObjectKeys = (obj, keys = []) => {
+  for (let key in obj) {
+    keys.push(key)
+    if (_.isObject(obj[key])) {
+      extractObjectKeys(obj[key], keys)
```

Comment:
If there are two nested objects with the same key, the key will be returned twice.
However later the call to `_.difference()` does not check if the key has been provided twice or not.

What about returning keys in a nested format, such as `some.nested.key`?

Reply:
can you share example of object that will fail?

Reply:
it should extract all keys, regardless if same and just knowing it'll not match is enough to ring the bell

Reply:
An example would be the following config:
```
{
  send: { fee: 1 },
  pay: { fee: 2 }
}
```
This particular config returns `[ 'send', 'fee', 'pay', 'fee' ]`.
When calling `_difference()`, it's not clear if `fee` has been provided twice or not.

Reply:
yes, but I we care is that if at least one fee will be missing in config vs example, we'll exit and user can then investigate what's missing. Having full path displayed would be more user friendly but it would be miles more complex to build vs simple recursion

Reply:
In this example, the validation will succeed, although it should not, I believe.

**config.example**
```
{
  send: { fee: 1 },
  pay: { fee: 2 }
}
```

**real.config**
```
{
  send: { fee: 1 },
  pay: {}
}
```

Since `pay.fee` is not configured, the validation shall fail I believe.
From the code it appears it will be successful. But I haven't run the code to double-check to be honest.